src/modules/commander/commander.cpp

```diff
@@ -2533,6 +2535,21 @@ int commander_thread_main(int argc, char *argv[])
 			}
 		}
 
+		/* reset mission state when disarmed */
+		if (!armed.armed && need_mission_reset) {
+			struct mission_s mission_tmp = {};
```

Comment:
This introduces a race with the mavlink app, which also publishes this topic. If we publish something from the commander, it would need to be done on a different topic.